### ISSUE TITLE
[BUG] Heavy load in cart because of load loop

### ISSUE THREAD

**Issue by machielkreiken** [tig-nl/postnl-magento2#241]
**To Reproduce**
When loading the shopping cart, the PostNL module checks whether an order can be a letterbox package. For this, it is checked per product whether the product meets the requirements.

For large orders, this check can even take up to 2 seconds, since all data is collected separately for each product instead of for all products at once.

<img width="1737" alt="Schermafbeelding 2021-03-10 om 11 29 21" src="https://user-images.githubusercontent.com/28918979/110914519-c60a1200-8316-11eb-91cc-692e3ab0889c.png">

The code can be found in

```
TIG\PostNL\Service\Shipping\LetterboxPackage.php    
public function isLetterboxPackage($products, $isPossibleLetterboxPackage)
    {
....

        foreach ($products as $product) {
            $this->fitsLetterboxPackage($product);
        }
....
    }

```

Within the `fitsLetterboxPackage` a repository call is made, which retrieves all data from a product, per product.

```
    public function fitsLetterboxPackage($productItem)
    {
        $maximumQtyLetterbox = $this->getMaximumQtyLetterbox($productItem);
...
    }

    /**
     * @param $productItem
     *
     * @return float
     * @throws \Magento\Framework\Exception\NoSuchEntityException
     */
    private function getMaximumQtyLetterbox($productItem)
    {
        $product = $this->productRepository->getById($productItem->getProductId());

        return floatval($product->getPostnlMaxQtyLetterbox());
    }

```
Even with a small shopping cart, the impact of the module can already be called great

<img width="1737" alt="Schermafbeelding 2021-03-10 om 11 59 44" src="https://user-images.githubusercontent.com/28918979/110914565-d4f0c480-8316-11eb-9e2d-f3a83ad2ebf0.png">

Solution direction

For all products in the quote, the required data can be retrieved in 1 collection. When this is done via a collection, only 1 query will go to the database instead of 1 per product. In addition, it is better to control in which data is collected, instead of loading complete products.

**Expected result**
1 call for all data

**Actual result**
call for every single product in cart

**Workaround**
None

**Screenshots**
See reproduction steps

** Please complete the following information**
- Magento version: [e.g. Magento 2.3.6]
- PHP version: [e.g. 7.2.0]
- TIG PostNL version: [e.g. 1.9]

**Comment by tig-jeffreybranderhorst** [comment]
Hi @machielkreiken ,

Thank you for submitting this, I will inform my colleagues, so we can investigate this further. Thank you for making the extension even better! 

Have a great day,
Jeffrey

**Comment by barryvdh** [comment]
We are experiencing a similar issue. The checkout becomes very slow with multiple products. In our case we're using Bundle products, so only a few are added to the cart but I think the logic happens for all child products as well? This make the checkout take 30+ seconds in some cases.

**Comment by dtcu8ey6p41w** [comment]
@tig-jeffreybranderhorst Hi Jeffery, we are also having the same issue, is there already a fix this problem?

**Comment by tig-vincentthart** [comment]
@barryvdh, @machielkreiken and @dtcu8ey6p41w 

This has been resolved in v1.9.8. If you still run into some issues please contact our support.

Have a nice day,

Vincent
